cherrypy/lib/static.py

```diff
@@ -314,6 +314,10 @@ def staticdir(section, dir, root='', match='', content_types=None, index='',
     branch = request.path_info[len(section) + 1:]
     branch = urllib.parse.unquote(branch.lstrip(r'\/'))
 
+    # On Windows, correct the path for long-path support
+    if os.name == 'nt':
+        branch = branch.replace('/', '\\')
```

Comment:
I think it's better to start relying on `pathlib`

Reply:
Pathlib is not available in Python 2.7 or anything below 3.4 so I don't think it applies right now.

Reply:
https://pypi.python.org/pypi/pathlib/

Reply:
You want to introduce a whole new dependency just for a 2 line change?

Reply:
Well, this just looks redundant to me, that's why I'm looking into using this package, which handles path processing way better than such simplistic things.

And I don't truly understand what is this `\\?` notation and why one would like to combine different delimiter styles. I just want this change, which I cannot check, to not reduce reliability.

Reply:
Agreed, Windows is weird. Only using this notation we can reliably get all paths (including those with unicode) to work on Windows. The mixing of delimiter styles is inherent to Cherrypy running on Windows, in that sense it's actually surprising that it worked so far. Windows users will supply paths in Windows style.
Unfortunately I have no experience with `pathlib` and combined with my lack of understanding of cherrypy internals, me implementing it would probably not work out well.
Simple isn't necessarily bad ;)

Reply:
Okay, let's postpone pathlib integration question, but I'll ask you to document the problem this change solves in code via comment.